projects/media-viewer/src/lib/media-viewer/viewers/pdf-viewer/pdf-viewer.component.ts

```diff
@@ -30,62 +33,51 @@ export class PdfViewerComponent implements AfterViewInit {
 
   @ViewChild('viewerContainer') viewerContainer: ElementRef;
 
-  pdfViewer: pdfjsViewer.PDFViewer;
-  pdfFindController: pdfjsViewer.PDFFindController;
+  private pdfViewer: PdfJsWrapper;
 
-  constructor(private pdfWrapper: PdfJsWrapper, private printService: PrintService) {}
+  constructor(
+    private readonly pdfJsWrapperFactory: PdfJsWrapperFactory,
+    private readonly printService: PrintService
+  ) {}
 
   async ngAfterViewInit(): Promise<void> {
-    [this.pdfViewer, this.pdfFindController] = await this.pdfWrapper.initViewer(this.url, this.viewerContainer);
-
-    this.pdfViewer.eventBus.on('updatefindcontrolstate', e => {
-      if (e.state === FindState.NOT_FOUND || e.state === FindState.FOUND) {
-        this.searchResults.next(e.matchesCount);
-      }
-    });
-    this.pdfViewer.eventBus.on('updatefindmatchescount', e => {
-      this.searchResults.next(e.matchesCount);
-    });
-    this.pdfViewer.eventBus.on('pagechanging', e => this.currentPageChanged.next(new SetCurrentPageOperation(e.pageNumber)));
+    this.pdfViewer = this.pdfJsWrapperFactory.create(this.viewerContainer);
```

Comment:
would it be clearer if this was called pdfWrapper

Reply:
agreed